src/orion/core/cli/db/upgrade.py

```diff
@@ -122,8 +122,13 @@ def upgrade_documents(storage):
     for experiment in storage.fetch_experiments({}):
         add_version(experiment)
         uid = experiment.pop("_id")
+        if "algorithms" in experiment:
```

Comment:
It seems unlikely but `algorithm` could potentially be undefined if the data is corrupted and there are no `algorithm` nor `algorithms` in `experiment`. We should handle this case gracefully. Like setting first `algorithm=None` and if its still None after the 2 ifs then we raise an error saying that the data was somehow corrupted as there were no algorithm(s) in the experiment.